### PR TITLE
Unify_certs: Modification in tls exchange to use new certs path

### PR DESCRIPTION
**Pull request by situ-s**
* **Problem Overview**  
  Key exchange and reconfiguration changes for new certificate scheme and changed path.
* **Testing Done**  
 Hermes test suites have passed with/without flag in private branch.
 Deployment tests are being done by Akhila. Will update here once all testings are completed.

### CODE REVIEW

**Comment by teoparvanov** [comment]
Hi @situ-s , is there a reason why the unit tests can't be included in this PR?
Without tests, how can you be confident that the proposed code changes work as intended?

CC: @yuliasherman @HristoStaykov

**Comment by situ-s** [comment]
> Hi @situ-s , is there a reason why the unit tests can't be included in this PR? Without tests, how can you be confident that the proposed code changes work as intended?
> 
> Also, as this change may impact certain reconfiguration scenarios of the product, maybe it's a good idea to create a private build and run existing Hermes tests on this build, such as those related to key exchange.
> 
> CC: @yuliasherman @HristoStaykov @arc-vmware

Hi @teoparvanov , As all the changes are under the flag useUnifiedCertificates, and this is disabled for now. This makes sure that changes will not cause any issue in existing system. I have manually tested the system by enabling the flag, but a lot more changes are needed for end-to-end flow. These changes along with Unit test will be covered in upcoming PRs.

**Comment by yuliasherman** [comment]
> > Hi @situ-s , is there a reason why the unit tests can't be included in this PR? Without tests, how can you be confident that the proposed code changes work as intended?
> > Also, as this change may impact certain reconfiguration scenarios of the product, maybe it's a good idea to create a private build and run existing Hermes tests on this build, such as those related to key exchange.
> > CC: @yuliasherman @HristoStaykov @arc-vmware
> 
> Hi @teoparvanov , As all the changes are under the flag useUnifiedCertificates, and this is disabled for now. This makes sure that changes will not cause any issue in existing system. I have manually tested the system by enabling the flag, but a lot more changes are needed for end-to-end flow. These changes along with Unit test will be covered in upcoming PRs.

@situ-s We are not going to merge those changes even under the flag until all required code modifications are completed and ETE testing (including deployment) with/without a feature flag is done.

File: kvbc/src/reconfiguration_kvbc_handler.cpp [hunk 1]
```diff
@@ -567,8 +567,11 @@ bool ReconfigurationHandler::handle(const concord::messages::ClientKeyExchangeSt
   for (const auto& gr : bftEngine::ReplicaConfig::instance().clientGroups) {
     for (auto cid : gr.second) {
       if (command.tls) {
-        std::string client_cert_path = bftEngine::ReplicaConfig::instance().certificatesRootPath + "/" +
-                                       std::to_string(cid) + "/client/client.cert";
+        std::string client_cert_path =
+            (bftEngine::ReplicaConfig::instance().useUnifiedCertificates)
```
Comment (yuliasherman) [inline kvbc/src/reconfiguration_kvbc_handler.cpp:571]: Nit: define const std::string base_path = bftEngine::ReplicaConfig::instance().certificatesRootPath;
to make an operator below shorter.

File: bftengine/src/bftengine/KeyExchangeManager.cpp [hunk 2]
```diff
@@ -145,10 +145,13 @@ void KeyExchangeManager::loadPublicKeys() {
 void KeyExchangeManager::exchangeTlsKeys(const std::string& type, const SeqNum& bft_sn) {
   auto keys = concord::util::crypto::Crypto::instance().generateECDSAKeyPair(
       concord::util::crypto::KeyFormat::PemFormat, concord::util::crypto::CurveType::secp384r1);
+  bool use_unified_certs = bftEngine::ReplicaConfig::instance().useUnifiedCertificates;
   std::string root_path =
-      bftEngine::ReplicaConfig::instance().certificatesRootPath + "/" + std::to_string(repID_) + "/" + type;
+      (use_unified_certs)
+          ? bftEngine::ReplicaConfig::instance().certificatesRootPath + "/" + std::to_string(repID_)
```
Comment (yuliasherman) [inline bftengine/src/bftengine/KeyExchangeManager.cpp:151]: Nit: define const std::string base_path = bftEngine::ReplicaConfig::instance().certificatesRootPath;
to make an operator below shorter.

**Comment by situ-s** [comment]
> > > Hi @situ-s , is there a reason why the unit tests can't be included in this PR? Without tests, how can you be confident that the proposed code changes work as intended?
> > > Also, as this change may impact certain reconfiguration scenarios of the product, maybe it's a good idea to create a private build and run existing Hermes tests on this build, such as those related to key exchange.
> > > CC: @yuliasherman @HristoStaykov @arc-vmware
> > 
> > 
> > Hi @teoparvanov , As all the changes are under the flag useUnifiedCertificates, and this is disabled for now. This makes sure that changes will not cause any issue in existing system. I have manually tested the system by enabling the flag, but a lot more changes are needed for end-to-end flow. These changes along with Unit test will be covered in upcoming PRs.
> 
> @situ-s We are not going to merge those changes even under the flag until all required code modifications are completed and ETE testing (including deployment) with/without a feature flag is done.

Sure @yuliasherman and @teoparvanov , We will do ETE deployment testing as well as private build testing for Hermes suites before merging it.

**Pull request closed without merging**